physics/GFS_rrtmg_pre.F90

```diff
@@ -1053,6 +1057,38 @@ subroutine GFS_rrtmg_pre_run (im, levs, lm, lmk, lmp, n_var_lndp,        &
          enddo
        enddo
 
+! --- add spp
```

Comment:
Lines 1060 to 1073 are not necessary. Wrap lines 1075-1089 in `if (do_spp)` and use `spp_wts_rad` directly. Remove `pattern_spp_rad`.

Reply:
Agree to remove. Will do! Thanks for the review.

Reply:
How is this handled in the other SPP code in CCPP?  We should keep things consistent with what's already in MYNN PBL, MYNN SFC, and Thompson MP.

Reply:
@JeffBeck-NOAA I think, for both MP and RRTMG, I will do what @climbfuji suggests. In these two schemes with these similar lines, I mimic what Laurie did for MYNN PBL and SFC, which I don't think is necessary for MP and RRTMG. For MYNN PBL or SFC, I guess the thing is probably like this: before Laurie's implementation went in, variables pattern_spp_* were originally used in the ad hoc subroutines of MYNN PBL and SFC, and to minimize code change on our side, Laurie passed spp_wts_* (ccpp variables) to these pattern_spp_* (local variables). @llpcarson may have a better answer.

Reply:
@climbfuji The corresponding changes for fv3atm have already been submitted to @llpcarson add_spp branch yesterday (https://github.com/llpcarson/fv3atm/pull/2). Not sure if you want to take a look. TY!

Reply:
@bluefinweiwei, if we can confirm that this change to Thompson MP won't change the results, then I'm fine with making that change there.  However, it sounds like this code style was necessary for MYNN SPP due to horizontal dimension conversions?

Reply:
For MYNN SPP - I guess so, again, need @llpcarson or @climbfuji to confirm. Okay, I think I can go ahead and update this for Thompson MP as well (perhaps a ncdiff for a b4b check), which will again be sent back to Laurie/Dom for a review.

Reply:
Team decision: Changes to pattern variables (use incoming argument directly) will be taken care of by Laurie after merge of Weiwei RRTMG PR (this will affect all SPP, and b4b tests will be carried out to ensure identical results)

Reply:
Removed "pattern_spp_*" and spp_wts_* remain in the code. @llpcarson @JeffBeck-NOAA